### PR TITLE
feat: use frontend-plugin-framework to provide a FooterSlot

### PR DESCRIPTION
**Pull request by brian-smith-tcril**
Note: This removes the ability to use `LOGO_POWERED_BY_OPEN_EDX_URL_SVG` to set the logo in the footer. This was directly using `process.env` and therefore did not support the more robust configuration methods provided by https://github.com/openedx/frontend-platform/blob/master/src/config.js. The footer component itself supports using `LOGO_TRADEMARK_URL` to set the logo (and supports the more robust configuration methods).

**What changed?**

- [ More in depth breakdown of changes ]
- [ Peripheral things that got changed ]
- [ etc... ]

**Developer Checklist**
- [ ] Test suites passing
- [ ] Documentation and test plan updated, if applicable
- [ ] Received code-owner approving review
- [ ] Bumped version number [package.json](../package.json)

**Testing Instructions**

[ How should a reviewer test this PR? ]

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

**Comment by arbrandes** [comment]
Mind taking a look at the tests?  Looks like we're running into some related failures.

**Comment by brian-smith-tcril** [comment]
@arbrandes I pushed a fix for the tests.

**Comment by arbrandes** [comment]
Holding off here for a https://github.com/openedx/frontend-slot-footer implementation.

File: src/App.jsx
Comment (arbrandes) [inline src/App.jsx:27]: Mind also removing the variable from https://github.com/openedx/frontend-app-gradebook/blob/master/.env.development#L10?

**Review by arbrandes — APPROVED** [review]
Good to merge once the comment is addressed!

**Pull request merged**
